pandas/tests/indexes/test_range.py

```diff
@@ -110,7 +110,7 @@ def test_constructor_same(self):
         result = RangeIndex(index)
         tm.assert_index_equal(result, index, exact=True)
 
-        with pytest.raises(TypeError):
+        with pytest.raises(ValueError):
```

Comment:
Can you check the error message via the `match` keyword here, and for the other instances where you updated this.  IIRC there's an open issue where we're trying to add this across the entire test suite.